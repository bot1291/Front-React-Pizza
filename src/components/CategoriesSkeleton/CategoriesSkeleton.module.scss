@import '../../styles/variables.scss';

.categories {
	display: grid;
	grid-template-columns: 1fr 1fr 1fr 1fr 1fr;
	grid-gap: 10px;
	margin-right: 25px;
}

.category {
	padding: 13px 30px;
	background-color: $skeleton-white;
	border-radius: 30px;
	border: 3px solid $skeleton-main-color;
	width: 188px;
	height: 52px;

	&.active {
		color: $skeleton-white;
		background-color: $skeleton-main-color;

		.title {
			background-color: $skeleton-white;
		}
	}
}

.title {
	width: 100%;
	height: 100%;
	background-color: $skeleton-main-color;
	border-radius: 10px;
}

@media screen and (max-width: 1480px) {
	.categories {
		grid-template-columns: 1fr 1fr 1fr 1fr;
	}
}
