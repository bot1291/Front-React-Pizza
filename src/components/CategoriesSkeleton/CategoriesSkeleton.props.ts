import { DetailedHTMLProps, HTMLAttributes } from 'react';

export interface CategoriesSkeletonProps
	extends DetailedHTMLProps<HTMLAttributes<HTMLDivElement>, HTMLDivElement> {
	count: number;
}
