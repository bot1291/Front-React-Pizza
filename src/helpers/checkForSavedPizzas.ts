import { IPizza } from '../interfaces/IPizza';

const checkForSavedPizzas = (pizzas: IPizza[]): IPizza[] => {
	if (localStorage.pizzas) {
		return pizzas.map((pizza) => {
			const replacementPizza: IPizza = JSON.parse(
				localStorage.pizzas
			).find((pizza: IPizza) => pizza.title === pizza.title);
			if (replacementPizza) {
				return replacementPizza;
			}
			return pizza;
		});
	}
	return pizzas;
};

export default checkForSavedPizzas;
