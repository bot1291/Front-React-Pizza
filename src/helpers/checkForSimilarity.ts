const checkForSimilarity = (string: string, search: string): boolean => {
	return string
		.toLowerCase()
		.split(' ')
		.join('')
		.includes(search.toLowerCase().split(' ').join(''));
};

export default checkForSimilarity;
