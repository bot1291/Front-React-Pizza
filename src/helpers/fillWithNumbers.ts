const fillWithNumbers = (count: number): number[] => {
	const numbers: number[] = [];
	for (let i = 0; i < count; i++) {
		numbers.push(i);
	}
	return numbers;
};

export default fillWithNumbers;
