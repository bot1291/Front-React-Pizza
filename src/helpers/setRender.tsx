import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { createReduxStore } from '../store/store';

// eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
const setRender = <T,>(Component: JSX.Element, store?: Record<string, T>) => {
	return render(
		<Provider store={createReduxStore(store)}>{Component}</Provider>
	);
};

export default setRender;
