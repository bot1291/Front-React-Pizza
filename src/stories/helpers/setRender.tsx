import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createReduxStore } from '../../store/store';

const setRender = <T,>(
	Component: JSX.Element,
	store?: Record<string, T>
): JSX.Element => {
	return (
		<MemoryRouter>
			<Provider store={createReduxStore(store)}>{Component}</Provider>
		</MemoryRouter>
	);
};

export default setRender;
