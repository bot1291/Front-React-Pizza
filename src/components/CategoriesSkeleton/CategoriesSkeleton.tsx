import styles from './CategoriesSkeleton.module.scss';
import { CategoriesSkeletonProps } from './CategoriesSkeleton.props';
import cn from 'classnames';
import { FC } from 'react';
import fillWithNumbers from '../../helpers/fillWithNumbers';

export const CategoriesSkeleton: FC<CategoriesSkeletonProps> = ({
	count,
	className,
	...props
}) => {
	return (
		<div
			data-testid="wrapper"
			className={cn(className, styles.categories)}
			{...props}>
			<div className={cn(styles.active, styles.category)}>
				<div className={styles.title} />
			</div>
			{fillWithNumbers(count).map((c) => (
				<div data-testid="category" key={c} className={styles.category}>
					<div className={styles.title} />
				</div>
			))}
		</div>
	);
};
