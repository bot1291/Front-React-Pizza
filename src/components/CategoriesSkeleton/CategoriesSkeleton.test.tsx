import setRender from '../../helpers/setRender';
import { CategoriesSkeleton } from './CategoriesSkeleton';
import { screen } from '@testing-library/dom';
import * as fillWithNumbers from '../../helpers/fillWithNumbers';

describe('CategoriesSkeleton', () => {
	it('Should follow the structure', () => {
		const spyFunc = jest.spyOn(fillWithNumbers, 'default');
		setRender(<CategoriesSkeleton count={4} />);
		const categoryElems = screen.getAllByTestId('category');
		const wrapperElem = screen.getByTestId('wrapper');
		expect(spyFunc).toHaveBeenCalled();
		expect(categoryElems).toHaveLength(4);
		expect(wrapperElem).toMatchSnapshot();
	});
});
