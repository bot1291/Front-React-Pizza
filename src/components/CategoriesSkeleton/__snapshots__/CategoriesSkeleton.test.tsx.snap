// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CategoriesSkeleton Should follow the structure 1`] = `
<div
  class="categories"
  data-testid="wrapper"
>
  <div
    class="active category"
  >
    <div
      class="title"
    />
  </div>
  <div
    class="category"
    data-testid="category"
  >
    <div
      class="title"
    />
  </div>
  <div
    class="category"
    data-testid="category"
  >
    <div
      class="title"
    />
  </div>
  <div
    class="category"
    data-testid="category"
  >
    <div
      class="title"
    />
  </div>
  <div
    class="category"
    data-testid="category"
  >
    <div
      class="title"
    />
  </div>
</div>
`;
